bootstore/src/schemes/v0/peer.rs

```diff
@@ -0,0 +1,810 @@
+    ///
+    /// This can only be successfully called when a node has been initialized,
+    /// either as initial member or learner who has learned its share.
+    pub async fn load_rack_secret(
+        &self,
+    ) -> Result<RackSecret, NodeRequestError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx.send(NodeApiRequest::LoadRackSecret { responder: tx }).await?;
+        let res = rx.await?;
+        res
+    }
+
+    /// Inform the node of currently known IP addresses on the bootstrap network
+    ///
+    /// These are generated from DDM prefixes learned by the bootstrap agent.
+    pub async fn load_peer_addresses(
+        &self,
+        addrs: BTreeSet<SocketAddrV6>,
+    ) -> Result<(), NodeRequestError> {
+        self.tx.send(NodeApiRequest::PeerAddresses(addrs)).await?;
+        Ok(())
+    }
+
+    /// Get the status of this node
+    pub async fn get_status(&self) -> Result<Status, NodeRequestError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx.send(NodeApiRequest::GetStatus { responder: tx }).await?;
+        let res = rx.await?;
+        Ok(res)
+    }
+
+    /// Shutdown the node's tokio tasks
+    pub async fn shutdown(&self) -> Result<(), NodeRequestError> {
+        self.tx.send(NodeApiRequest::Shutdown).await?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Status {
+    fsm_state: &'static str,
+    peers: BTreeSet<SocketAddrV6>,
+    connections: BTreeMap<Baseboard, SocketAddrV6>,
+    accepted_connections: BTreeSet<SocketAddrV6>,
+    negotiating_connections: BTreeSet<SocketAddrV6>,
+}
+
+/// A node in the bootstore protocol
+///
+/// This is the primary type for running the lrtq. There is one node running on
+/// each sled on the rack. Each node runs in a tokio task, with separate tokio
+/// tasks for each connection to other peer nodes. Nodes drive the lrtq protocol
+/// via control of an underlying  `Fsm`.
+pub struct Node {
+    config: Config,
+    fsm: Fsm,
+    peers: BTreeSet<SocketAddrV6>,
+    handle_unique_id_counter: u64,
+
+    // boolean set when a `NodeApiRequest::shutdown` is received
+    shutdown: bool,
+
+    // Connections that have been accepted, but where negotiation has not
+    // finished. At this point, we only know the client port of the connection,
+    // and so cannot identify it as a peer `Node`.
+    accepted_connections: BTreeMap<SocketAddrV6, AcceptedConnHandle>,
+
+    // Connections initiated from this node that have not yet completed
+    // handshakes via `Hello` and `Identify` messages.
+    //
+    // We only store the initiating (connecting) side, not accepting side here.
+    initiating_connections: BTreeMap<SocketAddrV6, PeerConnHandle>,
+
+    // Active connections participating in scheme v0
+    //
+    // This consists of both initiating and accepting connections that have
+    // completed handshake and are now logically equivalent.
+    established_connections: BTreeMap<Baseboard, PeerConnHandle>,
```

Comment:
Why is `established_connections` keyed on `Baseboard` while the other two are keyed on `SocketAddrV6`? Maybe add a comment about this.

Also can `initiating` and `accepted` connections be combined into a single map?

Reply:
`initiating` and `accepted` track addresses with different ports so they can't be merged. The port for `accepted` is the ephemeral port, whereas for `initiating` it is the server port. This is also why the types are different for the values. We learn the server port of the accepted side during handshake.